bool    check
---
string  mode
uint8   mode_index